src/ssl/support.cc

```diff
@@ -390,9 +390,35 @@ ssl_verify_cb(int ok, X509_STORE_CTX * ctx)
 }
 
 void
-Ssl::SetupVerifyCallback(Security::ContextPointer &ctx)
+Ssl::ConfigurePeerVerification(Security::ContextPointer &ctx, int flags)
 {
-    SSL_CTX_set_verify(ctx.get(), SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, ssl_verify_cb);
+    int mode;
+
+    if(flags & SSL_FLAG_DONT_VERIFY_PEER) {
+        debugs(83, 2, "SECURITY WARNING: Peer certificates are not verified for validity!");
```

Comment:
Security alerts should be at a high level (eg DBG_IMPORTANT for ones that are only warning) so they get seen.

I am keen for the DONT_VERIFY_PEER flag to be erased. It is already removed from the documentation about these listening ports. So now this code is changing please add a debug statement about the deprecation.
```suggestion
        debugs(83, DBG_IMPORTANT, "SECURITY WARNING: Peer certificates are not verified for validity!");
        debugs(83, DBG_IMPORTANT, "UPGRADE NOTICE: The DONT_VERIFY_PEER flag is deprecated. Remove the clientca= option to disable client certificates.");
```

Reply:
Can we please separate improvements & refactoring from the feature enhancement? This would be a whole lot simpler if the debug level and message fixes were done separately.